tests/plugins/test_remove_worker_metadata.py

```diff
@@ -0,0 +1,127 @@
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+from __future__ import print_function, unicode_literals
+
+import os
+
+from flexmock import flexmock
+
+from atomic_reactor.core import DockerTasker
+from atomic_reactor.constants import PLUGIN_REMOVE_WORKER_METADATA_KEY
+from atomic_reactor.inner import DockerBuildWorkflow
+from atomic_reactor.plugin import ExitPluginsRunner
+from atomic_reactor.util import ImageName
+from atomic_reactor.plugins.exit_remove_worker_metadata import (defer_removal)
+
+from tests.constants import MOCK_SOURCE, TEST_IMAGE, INPUT_IMAGE
+from tests.docker_mock import mock_docker
+import pytest
+
+
+class MockConfigMapResponse(object):
+    def __init__(self, data):
+        self.data = data
+
+    def get_data_by_key(self, key):
+        return self.data[key]
+
+
+class MockOSBS(object):
+    def __init__(self, config_map):
+        self.config_map = config_map
+
+    def delete_config_map(self, name):
+        return True
+
+
+class MockSource(object):
+
+    def __init__(self, tmpdir):
+        tmpdir = str(tmpdir)
+        self.dockerfile_path = os.path.join(tmpdir, 'Dockerfile')
+        self.path = tmpdir
+
+    def get_dockerfile_path(self):
+        return self.dockerfile_path, self.path
+
+
+class MockInsideBuilder(object):
+
+    def __init__(self):
+        mock_docker()
+        self.tasker = DockerTasker()
+        self.base_image = ImageName(repo='fedora', tag='25')
+        self.image_id = 'image_id'
+        self.image = INPUT_IMAGE
+        self.df_path = 'df_path'
+        self.df_dir = 'df_dir'
+
+        def simplegen(x, y):
+            yield "some\u2018".encode('utf-8')
+        flexmock(self.tasker, build_image_from_path=simplegen)
+
+    def get_built_image_info(self):
+        return {'Id': 'some'}
+
+    def inspect_built_image(self):
+        return None
+
+    def ensure_not_built(self):
+        pass
+
+
+def mock_workflow(tmpdir):
+    workflow = DockerBuildWorkflow(MOCK_SOURCE, TEST_IMAGE)
+    setattr(workflow, 'builder', MockInsideBuilder())
+    setattr(workflow, 'source', MockSource(tmpdir))
+    setattr(workflow.builder, 'source', MockSource(tmpdir))
+
+    return workflow
+
+
+@pytest.mark.parametrize('fragment_key', ['metadata.json', None])
+def test_remove_worker_plugin(tmpdir, caplog, fragment_key):
+    workflow = mock_workflow(tmpdir)
+
+    koji_metadata = {
+        'foo': 'bar',
+        'spam': 'bacon',
+    }
+    metadata = {'metadata.json': koji_metadata}
+
+    name = 'build-1-x86_64-md'
+    osbs = MockOSBS({name: metadata})
+    defer_removal(workflow, name, osbs)
+
+    name = 'build-1-ppc64le-md'
+    osbs = MockOSBS({name: metadata})
+    defer_removal(workflow, name, osbs)
+
+    runner = ExitPluginsRunner(
+        None,
+        workflow,
+        [{
+            'name': PLUGIN_REMOVE_WORKER_METADATA_KEY,
+            "args": {}
+        }]
+    )
+
+    (flexmock(osbs)
```

Comment:
No need to mock `delete_config_map` twice:
```
.should_call("delete_config_map")
.with_args('build-1-ppc64le-md')
.with_args('build-1-x86_64-md')
```

This would assume `ordered()` too

Reply:
The tests should also make sure a correct - worker-specific - `osbs` instance is used here

Reply:
ok

Reply:
We want each flexmock...should_call to be applied to the correct osbs instance. So something like:
```
for name in ('build-1-x86_64-md', 'build-1-ppc64le-md'):
    osbs = MockOSBS({name: metadata})
    (flexmock(osbs)
     .should_call("delete_config_map")
     .with_args(name)
     .once()
     .and_return(True))
```

The `.once()` is important to make sure it's really called.

Reply:
ok